pandas/tests/indexes/period/methods/test_astype.py

```diff
@@ -41,7 +41,12 @@ def test_astype_conversion(self):
         tm.assert_index_equal(result, expected)
 
         result = idx.astype(str)
-        expected = Index([str(x) for x in idx], name="idx", dtype=object)
+        if using_infer_string:
+            expected = Index(
+                [str(x) if x is not NaT else None for x in idx], name="idx", dtype="str"
+            )
+        else:
+            expected = Index([str(x) for x in idx], name="idx", dtype=object)
```

Comment:
I think these are all moving in the right direction, but was it ever considered a feature in the past for NaT to cast to string in that fashion? 

Just want to be careful and possibly document that in the transition guide, if that's a "feature" we released at some point

Reply:
As far as I know, this was considered a bug in the past (see https://github.com/pandas-dev/pandas/issues/25353#issuecomment-1344219762), but we never got around fixing this. And I think 3.0 / the new default str dtype, is the good opportunity to fix this.

(that issue is about NaN, but I think the same goes for NaT?)

But certainly this needs to be documented as one of the breaking changes (will add it to the list in https://github.com/pandas-dev/pandas/issues/59328)

Reply:
Nice find and agreed with the approach. Thanks!

Reply:
> But certainly this needs to be documented as one of the breaking changes (will add it to the list in https://github.com/pandas-dev/pandas/issues/59328)

Updated that issue